app/src/main/java/com/gbancarel/adoptyourpet/Activity/MainActivity.kt

```diff
@@ -30,12 +31,13 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         viewModel.liveData.observe(this, petFinderObserver)
+        val intent = Intent(applicationContext, SearchActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
```

Comment:
Généralement on fait plutôt `SearchActivity.newIntent()`

et dans SearchActivity.
```
object {
 fun newIntent(context: Context) = Intent(applicationContext, SearchActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
}
```

Reply:
T'as vrmt besoin de `Intent.FLAG_ACTIVITY_NEW_TASK` dans ton use case ? A priori non. Si oui pq ?

Reply:
oui j'ai besoin c'est le seule moyen de start une activity en dehors de la class

Reply:
Ça d'accord mais les flags des activités c'est pas obligatoire.
Si tu regardes la [doc](https://developer.android.com/reference/android/content/Intent) tu as pour FLAG_ACTIVITY_NEW_TASK => If set, this activity will become the start of a new task on this history stack
Je vois pas pq tu voudrais ça. 
Si tu fais juste `Intent(applicationContext, SearchActivity::class.java)` ça devrait suffir plutôt que `Intent(applicationContext, SearchActivity::class.java).setFlags(...`
Non ?

Reply:
Lis pt le fonctionnement des stack d'activity si t'es jamais tombé dessus si ça peut aider. https://developer.android.com/guide/components/activities/tasks-and-back-stack

Reply:
![image](https://user-images.githubusercontent.com/56867268/90409568-a032cc80-e0a9-11ea-96cc-7a544f5b2534.png)

si je fais comme tu dis voila l'erreur que j'obtiens